src/features/workout-builder/ui/add-exercise-modal.tsx

```diff
@@ -0,0 +1,160 @@
+"use client";
+
+import { useState, useEffect } from "react";
+import Image from "next/image";
+import { Plus, Loader2 } from "lucide-react";
+import { useQuery } from "@tanstack/react-query";
+import { ExerciseAttributeValueEnum } from "@prisma/client";
+
+import { useCurrentLocale, useI18n } from "locales/client";
+
+import { useWorkoutBuilderStore } from "../model/workout-builder.store";
+import { getExercisesByMuscleAction } from "../actions/get-exercises-by-muscle.action";
+
+interface AddExerciseModalProps {
+  isOpen: boolean;
+  onClose: () => void;
+  selectedEquipment: ExerciseAttributeValueEnum[];
+}
+
+interface ExerciseWithAttributes {
+  id: string;
+  name: string;
+  nameEn: string;
+  fullVideoImageUrl: string | null;
+  attributes: Array<{
+    attributeName: { name: string };
+    attributeValue: { value: string };
+  }>;
+}
+
+interface MuscleGroup {
+  muscle: ExerciseAttributeValueEnum;
+  exercises: ExerciseWithAttributes[];
+}
+
+export const AddExerciseModal = ({ isOpen, onClose, selectedEquipment }: AddExerciseModalProps) => {
+  const t = useI18n();
+  const locale = useCurrentLocale();
+  const [expandedMuscle, setExpandedMuscle] = useState<string | null>(null);
+  const { exercisesByMuscle, setExercisesByMuscle, setExercisesOrder, exercisesOrder } = useWorkoutBuilderStore();
+
+  const { data: muscleGroups, isLoading } = useQuery({
+    queryKey: ["exercises-by-muscle", selectedEquipment],
+    queryFn: async () => {
+      const result = await getExercisesByMuscleAction({ equipment: selectedEquipment });
+      if (result?.serverError) {
+        throw new Error(result.serverError);
+      }
+      return result?.data as MuscleGroup[];
+    },
+    enabled: isOpen && selectedEquipment.length > 0,
+  });
+
+  useEffect(() => {
+    const handleEsc = (e: KeyboardEvent) => {
+      if (e.key === "Escape") onClose();
+    };
+    if (isOpen) {
+      document.addEventListener("keydown", handleEsc);
+      return () => document.removeEventListener("keydown", handleEsc);
+    }
+  }, [isOpen, onClose]);
+
+  const handleAddExercise = (exercise: ExerciseWithAttributes, muscle: ExerciseAttributeValueEnum) => {
```

Comment:
When adding a new exercise, it should appear at the bottom of the list. That's where the "Add" button is, so it makes more sense UX wise to keep the flow intuitive